src/components/10-atoms/heading/index.js

```diff
@@ -0,0 +1,90 @@
+import { LitElement, html, css, unsafeCSS } from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map';
+
+/* eslint-disable import/no-extraneous-dependencies */
+import defineOnce from '../../../utils/define-once';
+import { applyDefaults } from '../../../utils/with-react';
+import styles from './index.scss';
+
+class AXAHeading extends LitElement {
+  static get tagName() {
+    return 'axa-heading';
+  }
+
+  static get styles() {
+    return css`
+      ${unsafeCSS(styles)}
+    `;
+  }
+
+  static get properties() {
+    // Define properties and types
+    return {
+      variant: { type: String },
+      rank: { type: Number },
+    };
+  }
+
+  constructor() {
+    super();
+    // this functions applies default values per type and verifies if
+    // the HTML attribute has been set before defining the custom element
+    applyDefaults(this);
+  }
+
+  render() {
+    const classes = classMap({
+      'a-heading--secondary': this.variant === 'secondary',
+    });
+
+    switch (this.rank) {
```

Comment:
Cant you write something like

`
          <h${rank} class="a-heading ${classes}">
            <slot></slot>
          </h${rank}>
        `

Reply:
I tried that (and similar ways) and it did not work, that's why I approached it like that. Somehow lit-html does not accept interception outside of attribute values. But I agree, I should maybe take another look at this.

Reply:
I can't make it to work.

Reply:
ok.